oap-server/server-library/library-datacarrier-queue/src/main/java/org/apache/skywalking/oap/server/library/datacarrier/consumer/MultipleChannelsConsumer.java

```diff
@@ -33,11 +34,17 @@ public class MultipleChannelsConsumer extends Thread {
     @SuppressWarnings("NonAtomicVolatileUpdate")
     private volatile long size;
     private final long consumeCycle;
+    // The flag to indicate whether the consumer thread should consume data.
+    @Setter
+    private volatile boolean consumeFlag = false;
+    // Whether the consumer thread should wait for notification to consume data.
+    private boolean notifiable = false;
```

Comment:
```suggestion
    private final boolean notifiable = false;
```

This should be immutable.